### ISSUE TITLE
Improving ATM interface

### ISSUE THREAD

**Issue by OgelGames** [mt-mods/atm#15]
ATMs currently require way too much clicking to deposit and withdraw money, especially when you want to exchange denominations.

I'd like it to include:
- An inventory slot to deposit money. The items would never actually enter the inventory, but instead be immediately deposited.
- A button to deposit all money in a player's inventory.
- A field to specify exactly how much MG you want to withdraw. Could also include an "all" button.
- Buttons to select the preferred denomination for withdraws.

When withdrawing the money would be split as needed, beginning with the preferred denomination. For example, if withdrawing 379MG with 10MG notes preferred, it would output 37x 10MG, 1x 5MG, and 4x 1MG.

Example formspec: (created with [`formspec_editor`](https://content.minetest.net/packages/Just_Visiting/formspec_editor/))

![image](https://github.com/mt-mods/atm/assets/48543043/32fd404f-dae3-4028-b603-bd705225fec0)

Not sure if this is the best way to do it, but it's the best idea I could come up with :)

**Comment by Wuzzy2** [comment]
I agree, a redesign of the formspec is a good idea. I think the Quit button can be removed because it's standard practice to not have one in formspecs.

I still think having 1-click buttons to insta-withdraw money (without typing) would be useful. Maybe one for each denomination.

Whatever suggestion you use, it should ideally be playtested to see how more convenient or inconvenient to use it really is.

This raises the question what the added benefit of the higher-tier ATM models will be, tho.

**Comment by OgelGames** [comment]
> I still think having 1-click buttons to insta-withdraw money (without typing) would be useful. Maybe one for each denomination.

Maybe some buttons to set the amount? Like 10, 100, 1000. Then it's only a few clicks.

> This raises the question what the added benefit of the higher-tier ATM models will be, tho.

Personally I think they should all be the same, with only the textures (and recipes) being different, then players would actually use the other models, as opposed to only using the highest tier.

**Comment by S-S-X** [comment]
> Personally I think they should all be the same, with only the textures (and recipes) being different, then players would actually use the other models, as opposed to only using the highest tier.

This. Different tiers, to keep stuff relevant, should ideally provide different functionality instead of being linear upgrade.
For example there could be different ATM that only works with coins but IMO *not* so that one works with both coins and notes while other works only with notes.
